cli/catalog-api-v1/src/lib.rs

```diff
@@ -0,0 +1,9 @@
+//! This module contains the generated OpenAPI client for the Catalog API.
+//!
+//! The client is generated from the OpenAPI spec in `openapi.json` using the `progenitor` crate.
+//! The spec is managed by the Catalog API team and is updated when the upstream API changes.
+
+mod client;
+pub use client::*;
+
+pub mod mock;
```

Comment:
Could we put this behind a feature flag so that it's not in the final binary?

Reply:
i guess this might end up in the production stuff anyway if we used it for mocking in integration tests, but until we've got that figured out i can flag it sure